src/settings.c

```diff
@@ -171,14 +177,19 @@ int git_libgit2_opts(int key, ...)
 		{
 			const char *file = va_arg(ap, const char *);
 			const char *path = va_arg(ap, const char *);
-			if (!SSL_CTX_load_verify_locations(git__ssl_ctx, file, path)) {
-				giterr_set(GITERR_NET, "SSL error: %s",
-					ERR_error_string(ERR_get_error(), NULL));
-				error = -1;
-			}
+			error = git_openssl_set_cert_file(file, path);
+		}
+#elif GIT_MBEDTLS
+		{
+			const char *file = va_arg(ap, const char *);
+			const char *path = va_arg(ap, const char *);
+			if (file)
+				error = git_mbedtls_set_cert_file(file, 0);
+			if (error && path)
+				error = git_mbedtls_set_cert_file(path, 0);
```

Comment:
Thinking about this more we probably don't want to fall back to loading a dir if loading the file failed since the caller chooses if they want to load a file or dir. Maybe this would be better? 

```
if (file) {
	error = git_mbedtls_set_cert_file(file, 0);
} else if (path) {
	error = git_mbedtls_set_cert_file(path, 1);
} else {
	giterr_set(GITERR_NET, "cannot set certificate locations: no file or path given");
	error = -1;
}
```